### PR TITLE
fix(server): proper asset sync

### LINKED ISSUE #9534: Since apk v1.105.0 login/upload possible but timeline won't refresh, keeps spinning
### The bug

Using the android mobile app from the playstore on a Samsung S23U: Since v1.105.0 I get weird errors where photos will get uploaded but not refreshed in the app. Tried to logout and in again via google oauth and mail/pw. But after relogin the timelime is building indefinitely. 

I also tried direct ip in LAN. It somehow helps switching between Wifi and mobile Data. After doing that I'm able to see the remote photos and can upload photos but the gallery won't refresh and the backup cloud icon is crossed out*. The uploaded photos can be seen in the Webui though. I used a secondary android phone (some rooted xiaomi with a custom rom) to install v1.105.0 from github and compare with v1.104.0. due to https://github.com/immich-app/immich/issues/9495#issuecomment-2113818459

v1.104.0. works perfectly on the secondary phone but  v1.105.0  shows the exact same issues as on the primary android phone.

*(after turning wifi off and waiting for mobile data, the "building timeline" vanishes and I can see photos but it seems to keep building in the background, down left "fotos" is still spinning)
d8349746-099b-4491-8151-5fcaefc1824b

update: I can access remote albums, which also show the correct backup status (only on server/on device and server) it seems to be mainly a problem with the timeline. I can search things and filter people/places.

Can't see any unusual in the docker logs, no warnings, errors, nothing: 

```
[Nest] 171  - 05/16/2024, 3:42:58 PM     LOG [ImmichServer] [EventRepository] Websocket Connect:    E6ynr_a3d_NhrmPbAAAi
[Nest] 171  - 05/16/2024, 3:42:58 PM     LOG [ImmichServer] [EventRepository] Websocket Disconnect: E6ynr_a3d_NhrmPbAAAi
[Nest] 171  - 05/16/2024, 3:42:59 PM     LOG [ImmichServer] [EventRepository] Websocket Connect:    1-_EQHAJTMkWn8RpAAAk
[Nest] 171  - 05/16/2024, 3:42:59 PM     LOG [ImmichServer] [EventRepository] Websocket Connect:    RX_mGm_exUzRw-6IAAAm
[Nest] 171  - 05/16/2024, 3:43:31 PM     LOG [ImmichServer] [EventRepository] Websocket Disconnect: RX_mGm_exUzRw-6IAAAm
[Nest] 171  - 05/16/2024, 3:43:33 PM     LOG [ImmichServer] [EventRepository] Websocket Connect:    BmbhCQSv4hVrdZlfAAAo
[Nest] 171  - 05/16/2024, 3:43:39 PM     LOG [ImmichServer] [EventRepository] Websocket Disconnect: BmbhCQSv4hVrdZlfAAAo
[Nest] 171  - 05/16/2024, 3:43:39 PM     LOG [ImmichServer] [EventRepository] Websocket Disconnect: 1-_EQHAJTMkWn8RpAAAk
[Nest] 171  - 05/16/2024, 3:43:39 PM     LOG [ImmichServer] [EventRepository] Websocket Disconnect: uN7sAk-oJzB0tUGHAAAe
[Nest] 171  - 05/16/2024, 3:43:39 PM     LOG [ImmichServer] [EventRepository] Websocket Disconnect: PN-L07X6bTtW-oBNAAAg
[Nest] 171  - 05/16/2024, 4:47:53 PM     LOG [ImmichServer] [EventRepository] Websocket Disconnect: KK_9G-sHe9W1-O3CAAAF
[Nest] 171  - 05/16/2024, 5:01:15 PM     LOG [ImmichServer] [EventRepository] Websocket Connect:    h9UxaZ-tza0X7gN2AAAq
[Nest] 171  - 05/16/2024, 5:01:16 PM     LOG [ImmichServer] [EventRepository] Websocket Disconnect: h9UxaZ-tza0X7gN2AAAq
[Nest] 244  - 05/16/2024, 5:16:04 PM     LOG [ImmichMicroservices] [MediaService] Successfully generated WEBP image preview for asset 152589db-e996-4db3-a0b9-f0faf3b49be3
[05/16/24 17:16:04] INFO     Setting                                            
                             'XLM-Roberta-Large-ViT-H-14__frozen_laion5b_s13b_b9
                             0k' execution providers to                         
                             ['CPUExecutionProvider'], in descending order of   
                             preference                                         
[05/16/24 17:16:04] INFO     Loading clip model                                 
                             'XLM-Roberta-Large-ViT-H-14__frozen_laion5b_s13b_b9
                             0k' to memory                                      
[05/16/24 17:16:04] INFO     Setting 'buffalo_l' execution providers to         
                             ['CPUExecutionProvider'], in descending order of   
                             preference                                         
[Nest] 244  - 05/16/2024, 5:16:04 PM     LOG [ImmichMicroservices] [MediaService] Successfully generated WEBP image thumbnail for asset 152589db-e996-4db3-a0b9-f0faf3b49be3
[Nest] 244  - 05/16/2024, 5:16:05 PM     LOG [ImmichMicroservices] [MediaService] Successfully generated WEBP image preview for asset f2009d45-a27c-48d6-b9a0-073b73bbfb54
[Nest] 244  - 05/16/2024, 5:16:05 PM     LOG [ImmichMicroservices] [MediaService] Successfully generated WEBP image thumbnail for asset f2009d45-a27c-48d6-b9a0-073b73bbfb54
[05/16/24 17:16:08] INFO     Loading facial recognition model 'buffalo_l' to    
                             memory                                             

```

The mobile logs show  Key 'currentUser' not found in Store, Error while getting remote assets | ApiException 401: {"message":"Invalid user token","error":"Unauthorized","statusCode":401} , PersonService        | Error while fetching curated people | ApiException 400: Socket operation failed: GET /person (Inner exception: Failed host lookup: 'immich.mydomain.com')

### The OS that Immich Server is running on

docker

### Version of Immich Server

v1.105.1

### Version of Immich Mobile App

v1.105.0

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
irrelevant
```

### Your .env content

```Shell
irrelevant
```

### Reproduction steps

```bash
1. install apk v1.105.0
2. try to open or login
3. gallery won't refresh or stuck on timeline building
```

### Relevant log output

```shell
2024-05-16 14:50:02.199042 | SEVERE   | ImmichErrorLogger    | PlatformDispatcher - Catch all | Key 'currentUser' not found in Store |
#0      Store.get (package:immich_mobile/entities/store.entity.dart:35)
#1      SyncService._syncRemoteAlbumsToDb (package:immich_mobile/services/sync.service.dart:297)
#2      SyncService.syncRemoteAlbumsToDb.<anonymous closure> (package:immich_mobile/services/sync.service.dart:66)
#3      new Future.sync (dart:async/future.dart:305)
#4      AsyncMutex.run.<anonymous closure> (package:immich_mobile/utils/async_mutex.dart:17)
#5      _RootZone.run (dart:async/zone.dart:1655)
#6      _FutureListener.handleWhenComplete (dart:async/future_impl.dart:205)
#7      Future._propagateToListeners.handleWhenCompleteCallback (dart:async/future_impl.dart:808)
#8      Future._propagateToListeners (dart:async/future_impl.dart:864)
#9      Future._completeWithValue (dart:async/future_impl.dart:643)
<asynchronous suspension>

2024-05-16 14:50:02.198689 | SEVERE   | AssetService         | Error while getting remote assets | ApiException 401: {"message":"Invalid user token","error":"Unauthorized","statusCode":401} |
#0      SyncApi.getFullSyncForUser (package:openapi/api/sync_api.dart:101)
<asynchronous suspension>
#1      AssetService._getRemoteAssets (package:immich_mobile/services/asset.service.dart:116)
<asynchronous suspension>
#2      SyncService._syncRemoteAssetsForUser (package:immich_mobile/services/sync.service.dart:241)
<asynchronous suspension>
#3      SyncService._syncRemoteAssetsFull (package:immich_mobile/services/sync.service.dart:231)
<asynchronous suspension>
#4      SyncService.syncRemoteAssetsToDb.<anonymous closure> (package:immich_mobile/services/sync.service.dart:56)
<asynchronous suspension>
#5      AsyncMutex.run.<anonymous closure> (package:immich_mobile/utils/async_mutex.dart:15)
<asynchronous suspension>

2024-05-16 13:54:33.162050 | SEVERE   | PersonService        | Error while fetching curated people | ApiException 400: Socket operation failed: GET /person (Inner exception: Failed host lookup: 'immich.mydomain.com')

#0      IOClient.send (package:http/src/io_client.dart:94)
<asynchronous suspension>
#1      BaseClient._sendUnstreamed (package:http/src/base_client.dart:93)
<asynchronous suspension>
#2      ApiClient.invokeAPI (package:openapi/api_client.dart:101)
<asynchronous suspension>
#3      PersonApi.getAllPeople (package:openapi/api/person_api.dart:103)
<asynchronous suspension>
#4      PersonService.getAllPeople (package:immich_mobile/services/person.service.dart:25)
<asynchronous suspension>
#5      getAllPeople (package:immich_mobile/providers/search/people.provider.dart:16)
<asynchronous suspension>
#6      FutureHandlerProviderElementMixin.handleFuture.<anonymous closure>.<anonymous closure> (package:riverpod/src/async_notifier/base.dart:348)
<asynchronous suspension>
 |
#0      ApiClient.invokeAPI (package:openapi/api_client.dart:104)
<asynchronous suspension>
#1      PersonApi.getAllPeople (package:openapi/api/person_api.dart:103)
<asynchronous suspension>
#2      PersonService.getAllPeople (package:immich_mobile/services/person.service.dart:25)
<asynchronous suspension>
#3      getAllPeople (package:immich_mobile/providers/search/people.provider.dart:16)
<asynchronous suspension>
#4      FutureHandlerProviderElementMixin.handleFuture.<anonymous closure>.<anonymous closure> (package:riverpod/src/async_notifier/base.dart:348)
<asynchronous suspension>

2024-05-16 13:54:33.161549 | SEVERE   | SearchService        | Failed to getExploreData | ApiException 400: Socket operation failed: GET /search/explore (Inner exception: Failed host lookup: 'immich.mydomain.com')

#0      IOClient.send (package:http/src/io_client.dart:94)
<asynchronous suspension>
#1      BaseClient._sendUnstreamed (package:http/src/base_client.dart:93)
<asynchronous suspension>
#2      ApiClient.invokeAPI (package:openapi/api_client.dart:101)
<asynchronous suspension>
#3      SearchApi.getExploreData (package:openapi/api/search_api.dart:90)
<asynchronous suspension>
#4      SearchService.getExploreData (package:immich_mobile/services/search.service.dart:116)
<asynchronous suspension>
#5      getPreviewPlacesProvider.<anonymous closure> (package:immich_mobile/providers/search/search_page_state.provider.dart:10)
<asynchronous suspension>
#6      FutureHandlerProviderElementMixin.handleFuture.<anonymous closure>.<anonymous closure> (package:riverpod/src/async_notifier/base.dart:348)
<asynchronous suspension>
 |
#0      ApiClient.invokeAPI (package:openapi/api_client.dart:104)
<asynchronous suspension>
#1      SearchApi.getExploreData (package:openapi/api/search_api.dart:90)
<asynchronous suspension>
#2      SearchService.getExploreData (package:immich_mobile/services/search.service.dart:116)
<asynchronous suspension>
#3      getPreviewPlacesProvider.<anonymous closure> (package:immich_mobile/providers/search/search_page_state.provider.dart:10)
<asynchronous suspension>
#4      FutureHandlerProviderElementMixin.handleFuture.<anonymous closure>.<anonymous closure> (package:riverpod/src/async_notifier/base.dart:348)
<asynchronous suspension>
```

### Additional information

I'm using the https://github.com/imagegenius/docker-immich monolythic dockerimage with tensorchord/pgvecto-rs:pg16-v0.2.0

### PR DESCRIPTION
**Pull request by fyfrey**
fixes #9534 and bunch of other issues

**breaking change! app and server must be on same version to work** 

with this change:

1. the server full sync method always returns the exact number of requested assets (unless it is the last chunk)
2. sync methods load the stacked asset count in a follow-up query via `loadRelationCountAndMap` so that `limit` works correctly
3. the app stops the fetching for more assets when the first non-full chunk is received
4. syncing uses `id` instead of `fileCreatedAt` because it may have weird metadata issues.

### CODE REVIEW

**Comment by alextran1502** [comment]
Cloning my prod instance tonight to test this in the morning! For science!!!

**Review by danieldietzler — APPROVED** [review]
Server code looks good to me!

File: server/src/queries/asset.repository.sql [hunk 1]
```diff
@@ -1049,49 +1049,18 @@ SELECT
...
-  AND ("stackedAssets"."deletedAt" IS NULL)
 WHERE
   "asset"."isVisible" = true
   AND "asset"."ownerId" IN ($1)
-  AND ("asset"."fileCreatedAt", "asset"."id") < ($2, $3)
+  AND ("asset"."createdAt", "asset"."id") < ($2, $3)
   AND "asset"."updatedAt" <= $4
 ORDER BY
-  "asset"."fileCreatedAt" DESC,
+  "asset"."createdAt" DESC,
```
Comment (mertalev) [inline server/src/queries/asset.repository.sql:1063]: This should have an index.

Comment (fyfrey) [reply server/src/queries/asset.repository.sql]: did not update the generated SQL, it's gone now

**Review by alextran1502 — APPROVED** [review]

**Pull request merged**